src/server/protocol.ts

```diff
@@ -1,24 +1,41 @@
 import type * as ts from "./_namespaces/ts";
 import type {
+    ApplicableRefactorInfo,
     CompilerOptionsValue,
+    CompletionsTriggerCharacter,
     EndOfLineState,
     FileExtensionInfo,
     HighlightSpanKind,
     InteractiveRefactorArguments,
-    MapLike,
-    OutliningSpanKind,
     OutputFile,
-    PluginImport,
-    ProjectReference,
+    RefactorTriggerReason,
+    RenameInfoFailure,
     RenameLocation,
     ScriptElementKind,
     ScriptKind,
+    SignatureHelpTriggerReason,
+    SymbolDisplayPart,
     TextChange,
     TextInsertion,
     TodoComment,
     TodoCommentDescriptor,
     TypeAcquisition,
+    UserPreferences,
 } from "./_namespaces/ts";
+import {
+    ClassificationType,
+    CompletionTriggerKind,
+    OrganizeImportsMode,
+    SemicolonPreference,
+} from "./_namespaces/ts";
+
+// These types/enums used to be defined in duplicate here and exported. They are re-exported to avoid breaking changes.
+export { ApplicableRefactorInfo, ClassificationType, CompletionsTriggerCharacter, CompletionTriggerKind, OrganizeImportsMode, RefactorTriggerReason, RenameInfoFailure, SemicolonPreference, SignatureHelpTriggerReason, SymbolDisplayPart, UserPreferences };
+
+type ChangeStringIndexSignature<T, NewStringIndexSignatureType> = { [K in keyof T]: string extends K ? NewStringIndexSignatureType : T[K]; };
+type ChangePropertyTypes<T, Substitutions extends { [K in keyof T]?: any; }> = {

```

Comment:
These will have to be exported for the API to be complete. (At least, that's what it looked like when I was testing the bundler fix).